Sources/LLBBuildSystem/Rules/RuleContext.swift

```diff
@@ -74,16 +78,22 @@ public class LLBRuleContext {
 
     private let artifactRoots: [String]
 
+    /// The function interface for evaluating requests.
```

Comment:
can you add a note here saying that this should only be used in rare and complex edge cases, and it's not expected to be used in most rule implementations? just for future reference for newcomers